mule-packager/src/main/java/org/mule/tools/api/util/MavenProjectInformation.java

```diff
@@ -62,14 +70,14 @@ public static MavenProjectInformation getProjectInformation(MavenSession session
       DistributionManagement management = project.getDistributionManagement();
       DeploymentRepository repository = management != null ? management.getRepository() : null;
       Settings settings = session.getSettings();
-      Optional<ExchangeRepositoryMetadata> metadata = getExchangeRepositoryMetadata(repository, settings);
+      Optional<ExchangeRepositoryMetadata> metadata = getExchangeRepositoryMetadata(repository, settings, customDomains);
       if (metadata.isPresent()) {
         if (deployments != null && isPlatformDeployment(deployments)) {
           builder.withDeployments(deployments);
         }
 
         // updating repository uri in order to deploy to Exchange
-        ExchangeRepositoryBuilder.updateRepositoryUri(project, session.getRequest().getSystemProperties());
+        new ExchangeHelper(customDomains).updateRepositoryUri(project, session.getRequest().getSystemProperties());
```

Comment:
Could we destructure it, to make it more readable?

ExchangeHelper exchangeHelper = new ExchangeHelper(customDomains);
var? systemProperties = session.getRequest().getSystemProperties();

and then just:
`exchangeHelper.updateRepositoryUri(project,systemProperties);`